core_acc_modules/gene_relationships.py

```diff
@@ -253,3 +253,94 @@ def get_relationship_in_expression_space(
     # total_counts = expression_dist_counts.groupby("offset")["total"].sum()[1]
     # expression_dist_counts["proportion"] = expression_dist_counts["total"]/total_counts
     return expression_dist_counts
+
+
+def find_related_acc_genes(
+    corr_df,
+    genes_to_consider,
+    gene_mapping_df,
+    num_top_genes,
+    operon_df=None,
+
+):
+
+    """For each gene in `genes_to_consider`, is the highest correlated, 2nd-highest correlated, 3rd
+    highest correlated gene, etc core or accessory? This function
+    finds the accessory genes that are highest correlated,
+    next highest correlated, third highest correlated, etc.
+    This function returns a dataframe with `genes_to_consider` as the index and the list
+    of accessory genes that they are most correlated with.
+
+    Arguments:
+    ----------
+    corr_df: pandas df
+        Dataframe containing the correlation matrix. The matrix is a gene x gene
+        dataframe containing values that correspond to how correlated each pair of
+        genes is
+    genes_to_consider: list
+        List of gene ids that will be used as starting genes
+    gene_mapping_df: pandas df
+        Dataframe with gene ids on the index and core/acc label as a column
+    num_top_genes: int
+        The number of correlated accessory genes to return
+    operon_df: pandas df
+        Dataframe with gene ids on the index and operon name as a column.
+        If this argument is provided, then genes within the same operon
+        will not be included in the counts.
+    """
+    # Get subset of genes
+    corr_subset = corr_df.loc[genes_to_consider]
+
+    # Note: PA14 contains duplicate rows so we will drop those here
+    corr_subset = corr_subset.drop_duplicates()
```

Comment:
Can't you just drop the duplicates from the `corr_df` so that there won't be duplicates in the subset and you don't have to rerun this line every time?

Reply:
Good point. I can pull this step out and move it to where I perform the correlation analysis

Reply:
Actually since this will require me to rerun many notebooks to verify the changes didn't distrub anything, which it shouldn't, I'll add this to my list of things to change when I refactor this repo